cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -1151,3 +1151,258 @@ def _validate(self, scheduler):
                 f"{scheduler} scheduler is not supported. Supported schedulers are: {', '.join(SUPPORTED_SCHEDULERS)}.",
                 FailureLevel.ERROR,
             )
+
+
+class _FlexibleInstanceTypesValidatorMixin:
+    def validate_property_homogeneity(
+        self,
+        instance_type_info_list: List[InstanceTypeInfo],
+        property_callback: Callable,
+        failure_message_callback: Callable[[Dict[str, int]], str],  # args: {instance_type: value, ...}
+        failure_level: FailureLevel,
+    ):
+        """Check if the instance_types have the same property (CPU count, GPU count etc)."""
+        instance_property = None
+        instance_type = None
+        for instance_type_info in instance_type_info_list:
+            current_instance_property = property_callback(instance_type_info)
+            current_instance_type = instance_type_info.instance_type()
+            if instance_property is not None and instance_property != current_instance_property:
+                mismatching_values = {
+                    instance_type: instance_property,
+                    current_instance_type: current_instance_property,
+                }
+                self._add_failure(failure_message_callback(mismatching_values), failure_level)
+                break
+            instance_property = current_instance_property
+            instance_type = current_instance_type
+
+
+class InstanceTypesListCPUValidator(Validator, _FlexibleInstanceTypesValidatorMixin):
+    """Confirm CPU requirements for Flexible Instance Types."""
+
+    def _validate(
+        self,
+        compute_resource_name: str,
+        instance_types_info: Dict[str, InstanceTypeInfo],
+        disable_simultaneous_multithreading: bool,
+        **kwargs,
+    ):
+        """Check if CPU requirements are met.
+
+        Instance types should have the same number of CPUs or same number of Cores if Simultaneous Multithreading
+        is disabled.
+        """
+        if disable_simultaneous_multithreading:
+            self.validate_property_homogeneity(
+                instance_type_info_list=list(instance_types_info.values()),
```

Comment:
1. 👍
2. I'm giving the caller of `validate_property_homogeneity` flexibility to use the mismatched values for now, since the wording hasn't been decided on yet.
3. 👍  I can set `ERROR` as the default argument for the `failure_level` parameter and avoid passing it explicitly.
4. I chose a callback to give the caller the flexibility to process the `instance_type_info` object